pkg/datastore/credentials_test.go

```diff
@@ -0,0 +1,19 @@
+package datastore
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewCredentialsProvider(t *testing.T) {
```

Comment:
can we add a fake-based test for the AWS IAM provider?

Reply:
added